smart_open/smart_open_lib.py

```diff
@@ -317,7 +370,7 @@ def _open_binary_stream(uri, mode, transport_params):
         return uri
 
     if not isinstance(uri, str):
-        raise TypeError("don't know how to handle uri %r" % uri)
+        raise TypeError("don't know how to handle uri %s" % repr(uri))
```

Comment:
Why this change?

Reply:
uri is a namedtuple, so using it with `%` is a type error

```
>>> '%r' % (1, 2, 3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: not all arguments converted during string formatting
```